addons/web/static/src/js/views/graph/graph_renderer.js

```diff
@@ -1,1099 +1,1046 @@
-    // Private
-    //--------------------------------------------------------------------------
+    function hexToRGBA(hex, opacity) {
+        const rgb = RGB_REGEX
+            .exec(hex)
+            .slice(1, 4)
+            .map(n => parseInt(n, 16))
+            .join(",");
+        return `rgba(${rgb},${opacity})`;
+    }
 
     /**
-     * This function aims to remove a suitable number of lines from the tooltip in order to make it reasonably visible.
-     * A message indicating the number of lines is added if necessary.
-     *
-     * @private
-     * @param {Number} maxTooltipHeight this the max height in pixels of the tooltip
+     * Used to avoid too long legend items.
+     * @param {string} label
+     * @returns {string} shortened version of the input label
      */
-    _adjustTooltipHeight: function (maxTooltipHeight) {
-        var sizeOneLine = this.$tooltip.find('tbody tr')[0].clientHeight;
-        var tbodySize = this.$tooltip.find('tbody')[0].clientHeight;
-        var toKeep = Math.floor((maxTooltipHeight - (this.$tooltip[0].clientHeight - tbodySize)) / sizeOneLine) - 1;
-        var $lines = this.$tooltip.find('tbody tr');
-        var toRemove = $lines.length - toKeep;
-        if (toRemove > 0) {
-            $lines.slice(toKeep).remove();
-            var tr = document.createElement('tr');
-            var td = document.createElement('td');
-            tr.classList.add('o_show_more');
-            td.innerHTML = _t("...");
-            tr.appendChild(td);
-            this.$tooltip.find('tbody').append(tr);
+    function shortenLabel(label) {
+        // string returned could be wrong if a groupby value contain a "/"!
+        const groups = label.split("/");
+        let shortLabel = groups.slice(0, 3).join("/");
+        if (shortLabel.length > 30) {
+            shortLabel = `${shortLabel.slice(0, 30)}...`;
+        } else if (groups.length > 3) {
+            shortLabel = `${shortLabel}/...`;
         }
-    },
-    /**
-     * This function creates a custom HTML tooltip.
-     *
-     * @private
-     * @param {Object} tooltipModel see chartjs documentation
-     */
-    _customTooltip: function (tooltipModel) {
-        this.$el.css({ cursor: 'default' });
-        if (this.$tooltip) {
-            this.$tooltip.remove();
+        return shortLabel;
+    }
+
+    class GraphRenderer extends OwlAbstractRenderer {
+        constructor() {
+            super(...arguments);
+
+            this.noDataLabel = [this.env._t("No data")];
+            this.sampleDataTargets = [".o_graph_canvas_container"];
+            this.config = this._prepareGraphConfig(this.props);
+
+            this.canvasRef = useRef("canvas");
+            this.containerRef = useRef("container");
         }
-        if (tooltipModel.opacity === 0) {
-            return;
+
+        async willUpdateProps(nextProps) {
+            this.config = this._prepareGraphConfig(nextProps);
```

Comment:
by consistency, call super here if you call it in mounted and patched (which is, imho, the correct thing to do)

Reply:
Done